core/src/main/scala/com/spotify/featran/FeatureSpec.scala

```diff
@@ -108,8 +110,45 @@ class FeatureSpec[T] private[featran] (private[featran] val features: Array[Feat
    * @param input input collection
    * @tparam M input collection type, e.g. `Array`, `List`
    */
-  def extract[M[_]: CollectionType](input: M[T]): FeatureExtractor[M, T] =
-    new FeatureExtractor[M, T](new FeatureSet[T](features, crossings), input, None)
+  def extract[M[_]: CollectionType](input: M[T]): FeatureExtractor[M, T] = {
+    import CollectionType.ops._
+
+    val fs = input.pure(featureSet)
+    new FeatureExtractor[M, T](fs, input, None)
+  }
+
+  /**
+   * Creates a new FeatureSpec with only the features that respect the given predicate.
+   *
+   * @param predicate Function determining whether or not to include the feature
+   */
+  def filter(predicate: Feature[T, _, _, _] => Boolean): FeatureSpec[T] =
+    new FeatureSpec[T](features.filter(predicate), crossings)
```

Comment:
Should we be filtering crossings here and other places?

Reply:
@richwhitjr indeed! I missed even after were talking about it lol!

Reply:
@richwhitjr I think this is what we need right?